portfolio/src/main/webapp/script.js

```diff
@@ -40,21 +40,26 @@ function addRandomStock() {
 }
 
 // Fetches comments from the server and adds them to the DOM.
-function LoadComments() {
-  const max = getMax();
-  fetch('/data?numCommentsToLoad=' + max).then(response => response.json()).then((comments) => {
-  const commentListElement = document.getElementById('comment-list');
-  commentListElement.innerHTML = ' ';
-  comments.forEach((comment) => {
-  commentListElement.appendChild(createListElement(comment));
+async function LoadComments() {
+  const loggedIn = await isLoggedIn();
+  if(loggedIn) {
+    const max = getMax();
+    fetch('/data?numCommentsToLoad=' + max).then(response => response.json()).then((comments) => {
+    const commentListElement = document.getElementById('comment-list');
+    commentListElement.innerHTML = ' ';
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createListElement(comment));
+      });
     });
-  });
+  }else{
```

Comment:
also small nit: there should be spaces around `} else {`

Reply:
addressed in "address nits"